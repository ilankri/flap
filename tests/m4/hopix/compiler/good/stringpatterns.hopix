fun f (x : string) : int =
  x ? { "luke" => 1 | "bulma" => 0 | "zelda" => 2 | (_ : string) => -1 }

fun check (x : int, expected : int) : unit =
  print_int (x);
  if x = expected then print_string (" OK\n") else print_string (" KO\n")

val test =
  check (f ("zelda"), 2);
  check (f ("bulma"), 0);
  check (f ("luke"), 1);
  check (f ("anakin"), -1)