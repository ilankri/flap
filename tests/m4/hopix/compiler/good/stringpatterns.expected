2 OK
0 OK
1 OK
-1 OK
